minari/dataset/minari_storage.py

```diff
@@ -1,25 +1,32 @@
 from __future__ import annotations
 
+import json
 import os
 import pathlib
-from collections import OrderedDict
-from typing import Any, Callable, Dict, Iterable, List, Optional, Tuple, Union
+import warnings
+from abc import ABC, abstractmethod
+from typing import Any, Callable, Dict, Iterable, List, Optional, Union
 
 import gymnasium as gym
-import h5py
 import numpy as np
 from gymnasium.envs.registration import EnvSpec
 
 from minari.serialization import deserialize_space, serialize_space
 
 
 PathLike = Union[str, os.PathLike]
+_METADATA_FILE_NAME = "metadata.json"
 
 
-class MinariStorage:
+class MinariStorage(ABC):
     """Class that handles disk access to the data."""
 
-    def __init__(self, data_path: PathLike):
+    def __init__(
+        self,
+        data_path: pathlib.Path,
+        observation_space: gym.Space,
+        action_space: gym.Space,
+    ):
```

Comment:
`__init__` is not supposed to be used by the user (it will throw an error as MinariStorage is an abstract class). I completely removed the docstring, so it doesn't appear in the documentation